src/complex/DataStructure/NeighborList.hpp

```diff
@@ -25,6 +25,8 @@ class NeighborList : public IDataArray
   using VectorType = std::vector<T>;
   using SharedVectorType = std::shared_ptr<VectorType>;
 
+  NeighborList() = default;
```

Comment:
And why, exactly, can't I make a stand alone DataArray or NeighborList?

Reply:
DataObject uses the DataStructure both directly and indirectly including within its constructor. Most DataObject operations either require either having or having had a DataStructure, which is why all constructors either require a DataStructure reference or copy values from an existing DataObject.

If you need a temporary DataObject, you can create a temporary DataStructure first that operates separately than the one data is being stored / accessed from. That way you avoid the possibility of existing observers being notified to temporary data being created and destroyed while also ensuring that temporary data is cleaned up at the end of the structure's scope without having to delete all temporary data manually like you would if you used the main DataStructure. It's one extra step to avoid a lot of misuse and confusion as to why data created that way isn't working like everything else is.